### ISSUE TITLE
PDF Export Errors with Raster Data

### ISSUE THREAD

**Issue by davidlgalt** [qgis/QGIS#30152]
I'm using QGIS 3.6.2 and this issue occurs when working with a low-resolution raster dataset in TIF format.

When refreshing a layout, there appears to be more than one draw. In the initial draw, everything is almost perfect, but there a few anomalies toward the middle of the map frame. After another second there's an adjustment and the map is as it should be.

When exporting to PDF or choosing the option to print to PDF it appears that the export occurs before the final draw as the anomaly from part way through the refresh process is present in the export. The same thing occurs when exporting to an image.

After much struggle, the cause of the error was discovered to be having a raster dataset in New York State Plane and the rest of the data and the map frame in  WGS 1984 (4326). 

The aesthetic of the map looks much better when the raster data and the map frame are not in the same projection, so I hope this can be fixed.

**Comment by gioman** [comment]
@davidlgalt please attach a sample project+data and exact steps to replicate (also some screenshot would help).

**Comment by davidlgalt** [comment]
OK. I was having trouble replicating the issue because I was wrong about one of the causes. The raster data and the project both have the same CRS. I tried 2 different coordinate systems (WGSW 1984 and New York StatePlane Central) and they both resulted in the same error.

[PDF_Export_Error.zip](https://github.com/qgis/QGIS/files/3279795/PDF_Export_Error.zip)

![QGISLayoutVsPDFExport](https://user-images.githubusercontent.com/44850852/59327193-ab1bf100-8c9d-11e9-9ba1-a48d2ae5f070.jpg)

**Comment by davidlgalt** [comment]
You also asked for the exact steps to replicate the error.

If you open the attached project, open the layout and choose the option to Export as PDF in the File Menu of the Layout Window.

Of note: the error occurs when in Layer Properties -> Symbology Resampling is set to Bilinear or Cubic. The error does not occur when Resampling is Nearest Neighbor (Default option).

**Comment by nirvn** [comment]
@nyalldawson , is that fixed via the disable tiled raster layer exports option introduced in 3.6?

**Comment by davidlgalt** [comment]
Hi Mathieu/Gioman,
The issue occurs when the disable tiled raster layer exports option is checked or unchecked. So far my only workaround is to take screenshots of layouts and use those to create PDFs. I have tried all sorts of other options but as long as I am using bilinear resampling the errors occur. Have you identified the issue? Can you suggest any better workarounds?

Thanks!

David

**Comment by davidlgalt** [comment]
I have been continuing to attempt to find a workaround for this issue. In my attempts, I have noticed that DPI and the chosen coordinate system affect when the error grid lines show up in the output.

For example, when I set the coordinate system as Google Maps Global Mercator (EPSG: 900913) the lines appear when the DPI is above 120.
 
When I set the coordinate system as NAD_1983_StatePlane_California_V_FIPS_0405_Feet
 (EPSG: ) the lines appear when the DPI is above 160.

Additionally, when I set the DPI higher (e.g. 600) many more error lines appear as in the screenshot below:



 
![image](https://user-images.githubusercontent.com/44850852/60741954-3d6f8780-9f20-11e9-95e6-81aa58201491.png)
